Source/SwiftLintFramework/Extensions/File+SwiftLint.swift

```diff
@@ -270,8 +270,11 @@ extension File {
 
     fileprivate func numberOfCommentAndWhitespaceOnlyLines(startLine: Int, endLine: Int) -> Int {
         let commentKinds = SyntaxKind.commentKinds
-        return syntaxKindsByLines[startLine...endLine].filter { kinds in
-            kinds.filter { !commentKinds.contains($0) }.isEmpty
+        let linesAndSyntaxKinds = zip(lines[(startLine - 1)...(endLine - 1)], syntaxKindsByLines[startLine...endLine])
```

Comment:
How about this?

```swift
return (startLine...endLine).filter { lineNumber in
    let line = lines[lineNumber - 1]
    let isLineEmpty = line.content.rangeOfCharacter(from: CharacterSet.whitespaces.inverted) == nil
    if isLineEmpty { return true }

    return syntaxKindsByLines[lineNumber].contains(where: SyntaxKind.commentKinds.contains)
}.count
```

It's simpler (IMHO) and more optimized because it avoids unconditionally iterating over all characters in the string, allocating a new string, and mutating it once for every whitespace character.

Reply:
Done